linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -429,14 +428,13 @@ final class Emitter(config: Emitter.Config) {
          * of the inlineable init, if there is one.
          */
         val ctorCache = classCache.getConstructorCache()
-        val ctorVersion = linkedInlineableInit.fold[Option[String]] {
-          linkedClass.version.map("1-" + _)
+        val ctorVersion = linkedInlineableInit.fold {
+          Version.combine(linkedClass.version)
         } { linkedInit =>
-          mergeVersions(linkedClass.version, linkedInit.version).map("2-" + _)
+          Version.combine(linkedClass.version, linkedInit.version)
```

Comment:
This is a bit funny. We can distinguish those two `Version`s because `Combine` of different lengths are always different. That's a bit of an implementation detail, though, isn't it? Also, what if we wanted the same distinguishing feature but coming from two different single `Version`s?

Should we have a `Version.tag(tag: Int, underlying: Version)` constructor for that purpose?

Reply:
I have considered that. IMO that is subsumed by:

```scala
Version.combine(Version.fromInt(tag), underlying)
```

(arguably that is *slighly* less space efficient, but I'm not sure it matters).

Reply:
Ah yes, that makes sense.